src/pool/srv_pool.c

```diff
@@ -7339,7 +7342,7 @@ pool_svc_update_map(struct pool_svc *svc, crt_opcode_t opc, bool exclude_rank,
 	bool				updated;
 	int				rc;
 	char				*env;
-	daos_epoch_t			rebuild_eph = d_hlc_get();
+	daos_epoch_t                     rebuild_eph = ds_rebuild_get_upbound_eph();
```

Comment:
Strictly speaking, it is still not safe to generate`rebuild_eph` via adding `epsilon` against current HLC before really refreshing the pool map. Because we do not know how long the refreshing pool map will take. If it takes more time than `epsilon`, then some new update may happen with newer epoch (than `rebuild_eph`) before refreshing pool map completed. And then current reintegration may not process those `new` data. That is unexpected.

On the other hand, since we have `epsilon` windows for reintegration, we need to double check VOS layer overwritten related logic to guarantee that if some data is `prepared` but not real `committed` yet, then the overwritten (from reintegration) with same epoch can be properly processed instead of returning failure.

Reply:
we may need some offline discuss.